src/main/kotlin/no/item/gradle/InitRunnerTask.kt

```diff
@@ -0,0 +1,22 @@
+package no.item.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.inject.Inject
+
+open class InitRunnerTask @Inject constructor(private val extension: InitRunnerExtension) : DefaultTask(){
+
+    @TaskAction
+    fun findXmlFiles() {
+        val list: MutableList<String> = ArrayList()
+        File(extension.fileDir).walk().forEach {
```

Comment:
I think using a `filter` and `map` would be nicer then mutating a list. It indicates intent more clearly then a `forEach`.

As far as possible, lets try to minimize the amount of mutation we do. Keep closer to a functional programming paradigm.

```kotlin
val files = File(extension.fileDir)
    .walk()
    .filter { it.isFile && it.extension == "xml" }
    .map { it.absolutePath }
```

Reply:
Good idea, implemented